tools/test/compileProtos.ts

```diff
@@ -115,6 +116,75 @@ describe('compileProtos tool', () => {
     assert(!ts.toString().includes('import * as $protobuf from "protobufjs"'));
   });
 
+  it('compiles protos to CJS and ES6 if --esm is specified', async function () {
+    this.timeout(20000);
+    await compileProtos.main([
+      '--esm',
+      path.join(__dirname, '..', '..', 'test', 'fixtures', 'esm'),
+    ]);
+    assert(fs.existsSync(expectedJsonResultFile));
+    assert(fs.existsSync(expectedJSResultFile));
+    assert(fs.existsSync(expectedTSResultFile));
+    assert(fs.existsSync(expectedCommonJSResultFile));
+
+    const json = await readFile(expectedJsonResultFile);
+    const root = protobuf.Root.fromJSON(JSON.parse(json.toString()));
+    assert(root.lookup('TestMessage'));
+    assert(root.lookup('LibraryService'));
+
+    const cjs = await readFile(expectedCommonJSResultFile);
+    assert(cjs.toString().includes('TestMessage'));
+    assert(cjs.toString().includes('LibraryService'));
+    assert(
+      cjs.toString().includes('http://www.apache.org/licenses/LICENSE-2.0')
+    );
+    assert(
+      cjs
+        .toString()
+        .includes('require("google-gax/build/src/protobuf").protobufMinimal')
+    );
+    assert(!cjs.toString().includes('require("protobufjs/minimal")'));
+
+    // check that it uses proper root object; it's taken from fixtures/package.json
+    assert(cjs.toString().includes('$protobuf.roots._org_fake_package'));
+
+    const js = await readFile(expectedJSResultFile);
+    assert(js.toString().includes('TestMessage'));
+    assert(js.toString().includes('LibraryService'));
+    assert(
+      js.toString().includes('http://www.apache.org/licenses/LICENSE-2.0')
+    );
+    assert(
+      js
+        .toString()
+        .includes(
+          'import { protobufMinimal  as $protobuf} from "google-gax/build/src/protobuf.js"'
+        )
+    );
+    assert(!js.toString().includes('require("protobufjs/minimal")'));
+    assert(
+      !js.toString().includes('import * as $protobuf from "protobufjs/minimal"')
+    );
+
+    // check that it uses proper root object; it's taken from fixtures/package.json
+    assert(js.toString().includes('$protobuf.roots._org_fake_package'));
+
+    const ts = await readFile(expectedTSResultFile);
+    assert(ts.toString().includes('TestMessage'));
+    assert(ts.toString().includes('LibraryService'));
+    assert(ts.toString().includes('import Long = require'));
+    assert(!ts.toString().includes('import * as Long'));
```

Comment:
Just curious - would `import Long from` work here?

Reply:
Good call out! I think it would, but I'm just looking to preserve functionality here, since I'm not making any changes to the .ts file.